sdk/python/feast/entity.py

```diff
@@ -29,17 +30,42 @@ def to_proto(self) -> EntityProto:
             Returns EntitySpec object
         """
         value_type = ValueTypeProto.ValueType.Enum.Value(self.dtype.name)
-        return EntityProto(name=self.name, value_type=value_type)
+        return EntityProto(
+            name=self.name,
+            value_type=value_type,
+            presence=self.presence,
+            group_presence=self.group_presence,
+            shape=self.shape,
+            value_count=self.value_count,
+            domain=self.domain,
+            int_domain=self.int_domain,
+            float_domain=self.float_domain,
+            string_domain=self.string_domain,
+            bool_domain=self.bool_domain,
+            struct_domain=self.struct_domain,
+            natural_language_domain=self.natural_language_domain,
+            image_domain=self.image_domain,
+            mid_domain=self.mid_domain,
+            url_domain=self.url_domain,
+            time_domain=self.time_domain,
+            time_of_day_domain=self.time_of_day_domain,
+        )
 
     @classmethod
-    def from_proto(cls, entity_proto: EntityProto):
+    def from_proto(cls, entity_proto: EntityProto, schema: schema_pb2.Schema = None):
         """
         Creates a Feast Entity object from its Protocol Buffer representation
 
         Args:
             entity_proto: EntitySpec protobuf object
+            schema: Schema from Tensorflow metadata, will be used to reference domain
+                defined at the schema level
 
         Returns:
             Entity object
         """
-        return cls(name=entity_proto.name, dtype=ValueType(entity_proto.value_type))
+        entity = cls(name=entity_proto.name, dtype=ValueType(entity_proto.value_type))
+        entity.update_presence_constraints(entity_proto)
+        entity.update_shape_type(entity_proto)
+        entity.update_domain_info(entity_proto, schema)
```

Comment:
Please move this schema out of the `from_proto` so that it is handled at the import_tfx_schema method.

Reply:
I removed it